pandas/_libs/groupby.pyx

```diff
@@ -1208,20 +1229,23 @@ def group_max(groupby_t[:, ::1] out,
                 if not _treat_as_na(val, True):
                     # TODO: Sure we always want is_datetimelike=True?
                     nobs[lab, j] += 1
-                    if val > maxx[lab, j]:
-                        maxx[lab, j] = val
+                    if compute_max:
+                        if val > group_min_or_max[lab, j]:
+                            group_min_or_max[lab, j] = val
+                    else:
+                        if val < group_min_or_max[lab, j]:
+                            group_min_or_max[lab, j] = val
```

Comment:
Wanted to be able to do something like if `PyObject_RichCompareBool(val, ..., cmp_method): ...` to simplify the conditional logic, but couldn't find a way to do something like this in Cython without the GIL.

Reply:
is there a perf impact here?  its checking compute_max at each step in a nested loop

Reply:
I didn't see any perf impact based on ASVs above. For a simple invariant like `if compute_max` where `compute_max` is defined outside the for loop, I'd guess this check is optimized away by the compiler. (It could also just be negligible compared to the cost of the indexing operations)

Reply:
makes sense.  worth it to de-duplicate a bunch of code.